src/main/java/org/springframework/samples/petclinic/owner/Owner.java

```diff
@@ -172,4 +172,16 @@ public void addVisit(Integer petId, Visit visit) {
 		pet.addVisit(visit);
 	}
 
+	public void updatePet(Pet newPet)
+	{
+		Pet existingPet = getPet(newPet.getId());
```

Comment:
imo, if there is no pet then some exception. The client want to update the pet, but what if no exists, then no info of that.